### PR TITLE
MSEARCH-34: Add holdings.discoverySuppress filter and facet property.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
## Changes:

* Add facet and filter for `holdings.discoverySuppress` property;
* Add `default` element to mappings that allow to set default value for a property when no property or null.

### CODE REVIEW

File: src/main/java/org/folio/search/utils/CollectionUtils.java [hunk 1]
```diff
@@ -24,4 +24,17 @@
...
+   *
+   * @param map - The map.
+   * @param key - Key to check.
+   * @param <K> - Key type
+   * @param <V> - Value type
+   * @return true if key has a non-null value associated, otherwise false.
+   */
+  public static <K, V> boolean hasNoValue(Map<K, V> map, K key) {
+    return map.get(key) == null;
+  }
```
Comment (hjiebsco) [inline src/main/java/org/folio/search/utils/CollectionUtils.java:39]: Not sure if a simple method like this is necessary.

File: src/main/resources/swagger.api/schemas/holding.json [hunk 2]
```diff
@@ -10,6 +10,10 @@
     "permanentLocationId": {
       "type": "string",
       "description": "The permanent shelving location in which an item resides."
+    },
+    "discoverySuppress": {
+      "type": "boolean",
+      "description": "records the fact that the record should not be displayed in a discovery system"
```
Comment (hjiebsco) [inline src/main/resources/swagger.api/schemas/holding.json:16]: Maybe change `records the fact` to `Indicates`?

**Review by hjiebsco — APPROVED** [review]

File: src/main/java/org/folio/search/utils/CollectionUtils.java (hunk 1 above)
Comment (bohdan-suprun) [reply src/main/java/org/folio/search/utils/CollectionUtils.java]: Removed and replaced with commons-MapUtils call.

File: src/main/resources/swagger.api/schemas/holding.json (hunk 2 above)
Comment (bohdan-suprun) [reply src/main/resources/swagger.api/schemas/holding.json]: Changed.

**Pull request merged**